### PR TITLE
Extract Aspire.Hosting.Nats.Tests project

### PR DESCRIPTION
**Pull request by Alirexaa**
Contributes to https://github.com/dotnet/aspire/issues/3185
Contributes to https://github.com/dotnet/aspire/issues/4294
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/5002)

### CODE REVIEW

File: tests/Aspire.Hosting.Nats.Tests/NatsFunctionalTests.cs [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+                                {
+                                    break;
+                                }
+                            }
+
+                            foreach (var item in events.Select((appEvent, i) => new { i, appEvent }))
+                            {
+                                Assert.Equal($"test-event-{item.i}", item.appEvent.Name);
+                                Assert.Equal($"test-event-description-{item.i}", item.appEvent.Description);
+                            }
```
Comment (eerhardt) [inline tests/Aspire.Hosting.Nats.Tests/NatsFunctionalTests.cs:206]: This should verify the expected number came through. We should convert it to `for (var i = 0; i < 10; i++)`. That way if less than 10 come through, the test fails.

File: tests/Aspire.Hosting.Nats.Tests/NatsFunctionalTests.cs [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+
+        using var host = hb.Build();
+
+        await host.StartAsync();
+
+        await pipeline.ExecuteAsync(async token =>
+        {
+            var natsConnection = host.Services.GetRequiredService<INatsConnection>();
+
+            var rtt = await natsConnection.PingAsync(token);
```
Comment (eerhardt) [inline tests/Aspire.Hosting.Nats.Tests/NatsFunctionalTests.cs:54]: Should we do something a little more than ping it? Can we send and consume a message?

Comment (Alirexaa) [reply tests/Aspire.Hosting.Nats.Tests/NatsFunctionalTests.cs:54]: I will do.

File: tests/Aspire.Hosting.Nats.Tests/NatsFunctionalTests.cs (hunk 1 above)
Comment (eerhardt) [reply tests/Aspire.Hosting.Nats.Tests/NatsFunctionalTests.cs:206]: That would help. But using a for loop would elimitate the Select and anonymous type that is happening here.

File: tests/Aspire.Hosting.Nats.Tests/NatsFunctionalTests.cs [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+                volumeName = VolumeNameGenerator.CreateVolumeName(nats1, nameof(WithDataShouldPersistStateBetweenUsages));
+
+                // if the volume already exists (because of a crashing previous run), try to delete it
+                DockerUtils.AttemptDeleteDockerVolume(volumeName);
+                nats1.WithDataVolume(volumeName);
+            }
+            else
+            {
+                bindMountPath = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+                nats1.WithDataBindMount(bindMountPath);
```
Comment (radical) [inline tests/Aspire.Hosting.Nats.Tests/NatsFunctionalTests.cs:88]: Just curious - `WithDataBindMount` is mounting to `/var/lib/nats` in the container. How is this test hitting that?

File: tests/Aspire.Hosting.Nats.Tests/NatsFunctionalTests.cs (hunk 2 above)
Comment (eerhardt) [reply tests/Aspire.Hosting.Nats.Tests/NatsFunctionalTests.cs:57]: I don't think this was resolved.

**Review by eerhardt — APPROVED** [review]
LGTM. Thanks again for the contributions!

**Comment by Alirexaa** [comment]
> LGTM. Thanks again for the contributions!

it's pleasure.

**Pull request merged**